### PR TITLE
add pinoy-hd.xyz provider

### PR DESCRIPTION
**Pull request by Jacekun**
- also renames pinoymoviepedia provider class file.

### CODE REVIEW

**Comment by LagradOst** [comment]
Do you want to merge this or keep working? is this a draft or not?

**Comment by Jacekun** [comment]
> Do you want to merge this or keep working? is this a draft or not?

Yes, it can be merged now.

**Comment by Jacekun** [comment]
Rebased on current changes

**Comment by Jacekun** [comment]
Uhh so yeah, final change.

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/PinoyHDXyzProvider.kt [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+            }
+            // Invoke sources
+            if (sources.isNotEmpty()) {
+                for (source in sources) {
+                    callback.invoke(source)
+                    //Log.i(this.name, "Result => (source) ${source.url}")
+                }
+                return true
+            }
+        } catch (e: Exception) {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/PinoyHDXyzProvider.kt:253]: try catch should be caught by a top level function, remove try catch here.

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/PinoyMoviesEsProvider.kt [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+            return all
+        } catch (e: Exception) {
+            e.printStackTrace()
+            Log.i(this.name, "Result => (Exception) ${e}")
+        }
+        return null
+    }
+    override fun getMainPage(): HomePageResponse {
+        val all = ArrayList<HomePageList>()
+        try {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/PinoyMoviesEsProvider.kt:72]: move down try catch

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/PinoyMoviesEsProvider.kt [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+    override fun loadLinks(
+            data: String,
+            isCasting: Boolean,
+            subtitleCallback: (SubtitleFile) -> Unit,
+            callback: (ExtractorLink) -> Unit
+    ): Boolean {
+        if (data == "about:blank") return false
+        if (data == "") return false
+        val sources = mutableListOf<ExtractorLink>()
+        try {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/PinoyMoviesEsProvider.kt:212]: remove try catch

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/PinoyMoviesEsProvider.kt [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+                                }
+                            }
+                            if (url.startsWith("https://voe.sx")) {
+                                val extractor = VoeExtractor()
+                                val src = extractor.getUrl(url)
+                                if (!src.isNullOrEmpty()) {
+                                    sources.addAll(src)
+                                }
+                            }
+                            if (url.startsWith("https://upstream.to")) {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/PinoyMoviesEsProvider.kt:238]: this should be moved to an extractor

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/PinoyMoviesEsProvider.kt [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+        if (data == "") return false
+        val sources = mutableListOf<ExtractorLink>()
+        try {
+            if (data.contains("playcontainer")) {
+                // parse movie servers
+                //Log.i(this.name, "Result => (data) ${data}")
+                val urls = Jsoup.parse(data).select("div")?.map { item ->
+                    item.select("iframe")?.attr("src")
+                }
+                if (!urls.isNullOrEmpty()) {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/PinoyMoviesEsProvider.kt:219]: use loadExtractor on urls

File: app/src/main/java/com/lagradost/cloudstream3/extractors/VoeExtractor.kt [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+    private data class ResponseLinks(
+        @JsonProperty("hls") val url: String?,
+        @JsonProperty("video_height") val label: Int?
+        //val type: String // Mp4
+    )
+
+    override fun getUrl(url: String, referer: String?): List<ExtractorLink> {
+        val extractedLinksList: MutableList<ExtractorLink> = mutableListOf()
+        val doc = Jsoup.parse(app.get(url).text)?.toString() ?: ""
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/extractors/VoeExtractor.kt:25]: ?: return listOf() or ?: trow, also why are you parsing it just to turn it into a string again wont just app.get(url).text work?

**Review by LagradOst — COMMENTED** [review]
some stuff can be changed

**Comment by LagradOst** [comment]
I get no links found on PinoyMoviesEsProvider, this might be due to the site "transferring files"

**Comment by LagradOst** [comment]
Do you want it merged or temporarily remove PinoyMoviesEsProvider from mainapi

**Comment by Jacekun** [comment]
> Do you want it merged or temporarily remove PinoyMoviesEsProvider from mainapi

Merge it and I'll just fix it later. Right now, series works and some movies with voe servers.

**Pull request merged**